### PR TITLE
Fix support for x86 builds with Clang 

### LINKED ISSUE #879: Error on x86 builds with Clang
Hello I'm encountering an issue while compiling with Clang targeting a x86 system

```
In file included from Generated Files\winrt/Windows.Foundation.h:5:
Generated Files\winrt/base.h(8299,9): error : no matching function for call to 'submit_threadpool_callback'
Generated Files\winrt/base.h(8222,17): message : candidate function not viable: no known conversion from '(lambda at Generated Files\winrt/base.h:8299:36)' to 'void (*)(void *, void *) __attribute__((stdcall))' for 1st argument
```

It looks suspiciously related to #496 and implementing a solution inspired from #511 I can successfully compile my project. 

However if I think I understand the problem (The Clang compiler lacks the Visual C++ support for lambdas with stdcall calling convention), I'm not familiar enough with the WinRT code base to understand the potential implications
I've attached the necessary modifications, I'll be happy to create a pull request if theses modifications make sense

[base_coroutine_threadpool.h.patch.txt](https://github.com/microsoft/cppwinrt/files/6116329/base_coroutine_threadpool.h.patch.txt)

### PR DESCRIPTION
**Pull request by arthurbiancarelli**
Fixes #879

### CODE REVIEW

**Comment by kennykerr** [comment]
FYI @oldnewthing this is fixing a regression introduced in #662.

File: strings/base_coroutine_threadpool.h [hunk 1]
```diff
@@ -85,21 +85,26 @@ namespace winrt::impl
...
+    struct threadpool_resume
+    {
+        threadpool_resume(com_ptr<IContextCallback> const& context, coroutine_handle<> handle) :
+            m_context(context), m_handle(handle) { }
+        com_ptr<IContextCallback> m_context;
+        coroutine_handle<> m_handle;
+    };
+
+    inline void __stdcall fallback_submit_threadpool_callback(void*,
+                                                            void* p) noexcept
```
Comment (kennykerr) [inline strings/base_coroutine_threadpool.h:97]:
```suggestion
    inline void __stdcall fallback_submit_threadpool_callback(void*, void* p) noexcept
```

Consistent formatting.

File: strings/base_coroutine_threadpool.h [hunk 2]
```diff
@@ -85,21 +85,26 @@ namespace winrt::impl
...
+    {
+        std::unique_ptr<threadpool_resume> state{ static_cast<threadpool_resume*>(
+            p) };
```
Comment (kennykerr) [inline strings/base_coroutine_threadpool.h:100]:
```suggestion
        std::unique_ptr<threadpool_resume> state{ static_cast<threadpool_resume*>(p) };
```

**Review by kennykerr — APPROVED** [review]
Thanks for the fix! Don't forget to complete the [CLA](https://github.com/microsoft/cppwinrt/pull/889#issuecomment-797571007).

**Comment by arthurbiancarelli** [comment]
Done

happy to contribute !

**Comment by oldnewthing** [comment]
Ah, I see. clang lambdas are implicitly convertible to `__cdecl` flat callbacks, but not `__stdcall`.

**Comment by kennykerr** [comment]
Yes, Visual C++ provides overloads for all calling conventions (a very cool feature) but this is apparently not required by the standard and Clang declines to do the same.

**Pull request merged**